src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -90,7 +90,9 @@ private static IOperation Unwrap(IOperation expression)
                     return;
                 }
 
-                if (invocation.Arguments.Length == 0)
+                if (invocation.Arguments.Length == 0 &&
+                    (invocation.TargetMethod.ContainingType.SpecialType == SpecialType.System_Object ||
+                    invocation.TargetMethod.OverriddenMethod?.ContainingType.SpecialType == SpecialType.System_Object))
```

Comment:
expression is too complex.  please don't mix/match &&'s and ||'s in this fashion.  just break into separate checks.

Reply:
Would

    if (invocation.Arguments.Length == 0)
    {
        var method = invocation.TargetMethod;
        while (method.OverriddenMethod != null)
        {
            method = method.OverriddenMethod;
        }

        if (method.ContainingType.SpecialType == SpecialType.System_Object)
        {
            // A call to `.ToString()` at the end of the interpolation.  This is unnecessary.
            // Just remove entirely.
            unwrapped = invocation.Instance;
            formatString = "";

            unnecessarySpans.AddRange(invocation.Syntax.Span
                .Subtract(invocation.Instance.Syntax.FullSpan));
            return;
        }
    }

be OK?

Reply:
If you want to be bulletproof, you could check the method name again because some languages (or at least the metadata) allows the override to use a different name. ToString is not the only overridable method on System.Object that can be called with no arguments.

Reply:
That's a new one for me. In theory then, could any parameterless method be overriding `ToString`, such that it should be checking only the name of the overriden method? (Is that a possibility we want to cover?).

Reply:
I'm not sure such corner cases are necessary especially how difficult it might be to test them. It doesn't make it a worse feature to fade overrides of ToString that aren't called ToString. But it is so niche that I'd be 100% fine with the IDE not picking up on that, too.

Reply:
i don't think you need the `if (invocation.Arguments.Length == 0) {` bit.  it's not going to be an override of object.ToString unless it has zero args.

Reply:
I guess `object.ToString` is unlikely to be overloaded any time soon, though I'd be tempted to leave it for clarity/explicitness. Removed for the time being.